### PR TITLE
test_engine_image_not_fully_deployed related cases

### PR DESCRIPTION
**Pull request by chriscchien**
Implement common function
- prepare_engine_not_fully_deployed_evnironment_with_volumes
- prepare_engine_not_fully_deployed_evnironment
- reset_nodes_taint
- restart_and_wait_ready_engine_count

Test case
- test_engine_image_miss_scheduled_perform_volume_operations

ref: longhorn [5242](https://github.com/longhorn/longhorn/issues/5242)

Signed-off-by: Chris Chien <chris.chien@suse.com>

### CODE REVIEW

File: manager/integration/tests/common.py [hunk 1]
```diff
@@ -3237,6 +3250,31 @@ def scale_up_engine_image_daemonset(client):
         assert e.status == 422
 
 
+def restart_and_wait_ready_engine_count(client, number): # NOQA
```
Comment (khushboo-rancher) [inline manager/integration/tests/common.py:3253]: Suggesting to change the param `number` to `ready_engine_count`

File: manager/integration/tests/test_ha.py [hunk 2]
```diff
@@ -2641,6 +2653,33 @@ def prepare_engine_not_fully_deployed_evnironment_with_volumes():
...
+    volume1.updateReplicaCount(replicaCount=2)
+
+    for r in volume1.replicas:
+        if r.hostId == nodes[0].id:
+            volume1.replicaRemove(name=r.name)
+            break
+
+    restart_and_wait_ready_engine_count(client, 2)
+
+    return volume1, volume2
```
Comment (khushboo-rancher) [inline manager/integration/tests/test_ha.py:2681]: Before returning the volume1, volume2, can we first fetch the latest volume object by `client.by_id_volume(volume_name)`

**Review by khushboo-rancher — CHANGES_REQUESTED** [review]

File: manager/integration/tests/test_ha.py (hunk 2 above)
Comment (chriscchien) [reply manager/integration/tests/test_ha.py:2684]: Hi @khushboo-rancher , updated!

**Comment by innobead** [comment]
For any PR, suggest having a corresponding issue instead to make the context clear.

**Comment by chriscchien** [comment]
Add test case `test_engine_image_miss_scheduled_perform_volume_operations`

**Comment by chriscchien** [comment]
Temporary close this PR for some problem encountered while adding test case. Will reopen and request review once clairfied problem

**Pull request closed without merging**